### PR TITLE
Timer Optimization

### PR DESCRIPTION
**Pull request by blogdaren**
make the master process timer API  just work like the children process timer：
```
//work under the master process
$tid1 = Timer::add(1, function(){
    //do sth
});
$tid2 = Timer::add(1, function(){
    //do sth
});

Timer::add(3, function($tid1){
    Timer::del($tid1);
}, [$tid1], false);

Timer::add(5, function(){
    Timer::delAll();
}, [], false);
```

### CODE REVIEW

File: Timer.php [hunk 1]
```diff
@@ -81,9 +100,10 @@ public static function signalHandle()
      * @param callable $func
      * @param mixed    $args
      * @param bool     $persistent
-     * @return int|false
+     * @param int      $persistent_timer_id
+     * @return int|bool
      */
-    public static function add($time_interval, $func, $args = array(), $persistent = true)
+    public static function add($time_interval, $func, $args = array(), $persistent = true, $persistent_timer_id = 0)
```
Comment (walkor) [inline Timer.php:106]: Thank you for your post. It is a good pr.
But `Timer::add` has a lot of parameters. I don't think we should add more parameters to it which will make the Timer more complicated to use. I would appreciate it if the parameter `persistent_timer_id` can be removed.

Comment (blogdaren) [reply Timer.php:106]: ok, i will find out whether it has a better workaround.

**Comment by blogdaren** [comment]
@walkor 
now i find out another workaround then need you to check whether it is acceptable:
1、keep all the `Timer` API remain unchanged including all the parameters;
2、the new workaround will use the internal `$task_args` to help complete this task,  here are the codes changed:

https://github.com/blogdaren/Workerman/commit/03e557a39d8a2378131708279b27d777299ff7d6

https://github.com/blogdaren/Workerman/commit/c8f9f340e569d885ffa9e349677b32ea173dbafe

thus it won't destroy any `Timer` API any more and also work well.
so if it is acceptable then i will issue a new PR, tks.

File: Timer.php [hunk 2]
```diff
@@ -163,7 +193,13 @@ public static function del($timer_id)
             return self::$_event->del($timer_id, EventInterface::EV_TIMER);
         }
 
-        return false;
+        foreach(self::$_tasks as $run_time => $task_data) 
+        {
+            if(array_key_exists($timer_id, $task_data))     unset(self::$_tasks[$run_time][$timer_id]);
+            if(array_key_exists($timer_id, self::$_status)) unset(self::$_status[$timer_id]);
```
Comment (walkor) [inline Timer.php:199]: I think 
```php
if(array_key_exists($timer_id, self::$_status)) unset(self::$_status[$timer_id]);
```
should move to outside of foreach.

Comment (blogdaren) [reply Timer.php:199]: yeah, u r right, tks.

**Pull request closed without merging**